### PR TITLE
Log empty pools

### PR DESCRIPTION
**Pull request by sschneid**
Make a note in the logfile when a pool is detected to be empty.

Also:

- vmpooler__empty__<pool> Redis key to determine when to log
- lifetime/TTL checks moved to `_check_running_vm` method
  - no longer pay attention to VMware-based 'host.runtime.bootTime'

This PR implements a bunch of other stuff to account for rspec testing:

- Thread creation and looping in `check_pool`
- Everything else in `_check_pool`

### CODE REVIEW

File: lib/vmpooler/pool_manager.rb [hunk 1]
```diff
@@ -289,161 +289,168 @@ def destroy_vm(vm, pool)
...
-          $redis.smembers('vmpooler__ready__' + pool['name']).each do |vm|
-            if inventory[vm]
-              begin
-                check_ready_vm(vm, pool['name'], pool['ready_ttl'] || 0)
-              rescue
-              end
+      # RUNNING
+      $redis.smembers('vmpooler__running__' + pool['name']).each do |vm|
+        if inventory[vm]
+          if pool['running_ttl']
```
Comment (colinPL) [inline lib/vmpooler/pool_manager.rb]: Can this change between loops? If not, can this passed as an argument to `_check_pool` and set `'720'` as the default value? This `if` can then be collapsed.

Comment (colinPL) [inline lib/vmpooler/pool_manager.rb]: Below, it looks like this `if` is collapsed into a `pool['ready_ttl'] || 0` statement. Can this be done here as well?

File: lib/vmpooler/pool_manager.rb [hunk 2]
```diff
@@ -289,161 +289,168 @@ def destroy_vm(vm, pool)
...
 
-            if inventory[vm]
-              begin
-                check_pending_vm(vm, pool['name'], pool['timeout'])
-              rescue
-              end
-            end
+      # PENDING
+      $redis.smembers('vmpooler__pending__' + pool['name']).each do |vm|
+        unless pool['timeout']
```
Comment (colinPL) [inline lib/vmpooler/pool_manager.rb]: `pool['timeout']` is not used unless the VM is present in `inventory`. This should be moved within the `if inventory[vm]` block.

**Comment by sschneid** [comment]
@colinPL New commit addresses your issues.

**Comment by sschneid** [comment]
Deployed on vmpooler-dev.

**Comment by sschneid** [comment]
This PR has been updated and is ready for re-review.

**Comment by rji** [comment]
It looks like this will need a rebase now that #114 has been merged

**Comment by sschneid** [comment]
Rebased.

**Comment by rji** [comment]
:+1:

**Pull request merged**